src/timedate/timedated.c

```diff
@@ -886,7 +887,7 @@ static int method_set_time(sd_bus_message *m, void *userdata, sd_bus_error *erro
         log_struct(LOG_INFO,
                    "MESSAGE_ID=" SD_MESSAGE_TIME_CHANGE_STR,
                    "REALTIME="USEC_FMT, timespec_load(&ts),
-                   LOG_MESSAGE("Changed local time to %s", ctime(&ts.tv_sec)));
+                   LOG_MESSAGE("Changed local time to %s", format_timestamp(buf, sizeof(buf), timespec_load(&ts))));
```

Comment:
format_timestamp() can return NULL if it doesn't like the timestamp. I think it needs to be wrapped in strnull() or strna().